src/Tribe/Attendees.php

```diff
@@ -752,72 +754,97 @@ public function sanitize_csv_value( $value ) {
 	 * @since 4.6.2
 	 *
 	 */
-	public function maybe_generate_csv() {
-		if ( empty( $_GET['attendees_csv'] ) || empty( $_GET['attendees_csv_nonce'] ) || empty( $_GET['event_id'] ) ) {
+	public function maybe_generate_csv( $all = false ) {
+		if ( empty( $_GET['attendees_csv'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
 			return;
 		}
 
-		$event_id = absint( $_GET['event_id'] );
-
-		$event_id = Event::filter_event_id( $event_id, 'attendee-csv-report' );
-
-		// Verify event ID is a valid integer and the nonce is accepted.
-		if ( empty( $event_id ) || ! wp_verify_nonce( $_GET['attendees_csv_nonce'], 'attendees_csv_nonce' ) ) {
+		if ( empty( $_GET['attendees_csv_nonce'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
```

Comment:
Looking at these checks, I feel like there are 2 things that would make this a bit better:

1. Have a dedicated function to check the various `$_GET` values, and trigger an early return if we determine we don't have what we need. This would help make the CSV logic easier to read and understand.
2. The nonce check should also happen as part of the above, rather than farther down the function.